Source/Core/Core/IOS/ES/ES.cpp

```diff
@@ -20,25 +20,49 @@
 #include "Common/Logging/Log.h"
 #include "Common/MsgHandler.h"
 #include "Common/NandPaths.h"
+#include "Core/Boot/Boot.h"
 #include "Core/Boot/Boot_DOL.h"
 #include "Core/ConfigManager.h"
+#include "Core/HLE/HLE.h"
 #include "Core/HW/DVDInterface.h"
 #include "Core/HW/Memmap.h"
 #include "Core/IOS/ES/ES.h"
 #include "Core/IOS/ES/Formats.h"
+#include "Core/PatchEngine.h"
+#include "Core/PowerPC/PPCSymbolDB.h"
 #include "Core/PowerPC/PowerPC.h"
 #include "Core/WiiRoot.h"
 #include "Core/ec_wii.h"
 #include "DiscIO/NANDContentLoader.h"
 #include "DiscIO/Volume.h"
+#include "VideoCommon/HiresTextures.h"
 
 namespace IOS
 {
 namespace HLE
 {
 namespace Device
 {
-std::string ES::m_ContentFile;
+// Shared across all ES instances.
+static std::string s_content_file;
+static std::vector<u64> s_title_ids;
+static struct TitleContext
```

Comment:
Please keep structure definitions/declarations and instance variable declarations separate from one another.

i.e.

```cpp
struct Thing
{
...
};
Thing s_thing;
```
instead of joining them together.